exporter/rabbitmqexporter/integration_test.go

```diff
@@ -0,0 +1,162 @@
+	password = "mypass"
+	vhost    = "myvhost"
+)
+
+func TestExportWithNetworkIssueRecovery(t *testing.T) {
+	testCase := []struct {
+		name  string
+		image string
+	}{
+		{
+			name:  "test rabbitmq latest",
+			image: "rabbitmq:latest",
+		},
+	}
+
+	for _, c := range testCase {
+		t.Run(c.name, func(t *testing.T) {
+			port := randPort()
+			req := testcontainers.ContainerRequest{
+				Image:        c.image,
+				ExposedPorts: []string{fmt.Sprintf("%s:5672", port)},
+				WaitingFor: wait.ForListeningPort("5672").
+					WithStartupTimeout(2 * time.Minute),
+				Env: map[string]string{
+					"RABBITMQ_DEFAULT_USER":  username,
+					"RABBITMQ_DEFAULT_PASS":  password,
+					"RABBITMQ_DEFAULT_VHOST": vhost,
+				},
+			}
+			container := getContainer(t, req)
+			defer func() {
+				err := container.Terminate(context.Background())
+				require.NoError(t, err)
+			}()
+
+			// Connect to rabbitmq then create a queue and queue consumer
+			host, err := container.Host(context.Background())
+			require.NoError(t, err)
+			endpoint := fmt.Sprintf("amqp://%s:%s", host, port)
+			connection, channel, consumer := setupQueueConsumer(t, logsRoutingKey, endpoint)
+
+			// Create and start rabbitmqexporter
+			factory := NewFactory()
+			cfg := factory.CreateDefaultConfig().(*Config)
+			cfg.Connection.Endpoint = endpoint
+			cfg.Connection.VHost = vhost
+			cfg.Connection.Auth = AuthConfig{Plain: PlainAuth{Username: username, Password: password}}
+			exporter, err := factory.CreateLogsExporter(context.Background(), exportertest.NewNopCreateSettings(), cfg)
+			require.NoError(t, err)
+			err = exporter.Start(context.Background(), componenttest.NewNopHost())
+			require.NoError(t, err)
+			defer func() {
+				err = exporter.Shutdown(context.Background())
+				require.NoError(t, err)
+			}()
+
+			// Export and verify data is consumed
+			logs := testdata.GenerateLogsOneLogRecord()
+			err = exporter.ConsumeLogs(context.Background(), logs)
+			require.NoError(t, err)
+			consumed := <-consumer
+			require.NotNil(t, consumed.Body)
+
+			// Stop the container before exporting the next logs to simulate a network issue
+			err = channel.Close()
+			require.NoError(t, err)
+			err = connection.Close()
+			require.NoError(t, err)
+			stopTimeout := time.Second * 5
+			err = container.Stop(context.Background(), &stopTimeout)
+			require.NoError(t, err)
+			logs = testdata.GenerateLogsOneLogRecord()
+			err = exporter.ConsumeLogs(context.Background(), logs)
+			require.Error(t, err)
+
+			// Restart container to simulate network issue recovery
+			err = container.Start(context.Background())
+			require.NoError(t, err)
+			connection, channel, consumer = setupQueueConsumer(t, logsRoutingKey, endpoint)
+			defer func() {
+				channel.Close()
+				connection.Close()
+			}()
+
+			logs = testdata.GenerateLogsOneLogRecord()
+			err = exporter.ConsumeLogs(context.Background(), logs)
+			require.NoError(t, err)
+			consumed = <-consumer
+			require.NotNil(t, consumed.Body)
```

Comment:
Would there be some "simple" way to check the container to see if it got the exported data as expected? Just a suggestion, it may be out of scope for this test. 👍

Reply:
Yeah that'd be a better test. Right now its consuming the message from the container. I just pushed some more code to unmarshal the message and verify it matches the input